crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -9,17 +8,17 @@ pub struct BlobVec {
     item_layout: Layout,
     capacity: usize,
     len: usize,
-    data: UnsafeCell<NonNull<u8>>,
-    swap_scratch: UnsafeCell<NonNull<u8>>,
+    data: NonNull<u8>,
+    swap_scratch: NonNull<u8>,
```

Comment:
I can confirm this is unnecessary. Neither pointer is itself changed behind `&self`. Only the pointed to value can be changed using `&self`, which is fine as they are raw pointers that are not derived from a reference.